### PR TITLE
Angular 15

### PR DESCRIPTION
**Pull request by withinfocus**
## Type of change

- [ ] Bug fix
- [ ] New feature development
- [X] Tech debt (refactoring, code cleanup, dependency upgrades, etc)
- [ ] Build/deploy pipeline (DevOps)
- [ ] Other

## Objective

Upgrades to Angular 15. Adds an Angular workspace and makes other core configuration changes along with the upgrade.

## Code changes

- **package.json:** Tons of package updates.

## Testing requirements

Full regression testing will be needed when this is mixed in with other updates.

## Before you submit

- [ ] I have checked for **linting** errors (`npm run lint`) (required)
- [ ] I have added **unit tests** where it makes sense to do so (encouraged but not required)
- [ ] This change requires a **documentation update** (notify the documentation team)
- [ ] This change has particular **deployment requirements** (notify the DevOps team)

### CODE REVIEW

File: tsconfig.json [hunk 1]
```diff
@@ -1,26 +1,27 @@
...
     "paths": {
       "tldjs": ["jslib/src/misc/tldjs.noop"],
       "jslib-common/*": ["jslib/common/src/*"],
       "jslib-angular/*": ["jslib/angular/src/*"],
       "jslib-electron/*": ["jslib/electron/src/*"],
       "jslib-node/*": ["jslib/node/src/*"]
-    }
-  },
-  "angularCompilerOptions": {
-    "preserveWhitespaces": true
```
Comment (Hinton) [inline tsconfig.json:23]: We still have this on the other clients.

**Review by Hinton — APPROVED** [review]
Seems good.

File: tsconfig.json (hunk 1 above)
Comment (withinfocus) [reply tsconfig.json:23]: Sure do, and the inheritance here confused me. Restored.

**Review by Hinton — DISMISSED** [review]
Seems good.

**Review by kspearrin — APPROVED** [review]

**Pull request merged**